packages/astro/src/core/build/plugins/util.ts

```diff
@@ -63,13 +74,34 @@ export function getVirtualModulePageNameFromPath(virtualModulePrefix: string, pa
 }
 
 /**
- *
- * @param virtualModulePrefix
- * @param id
+ * From the VirtualModulePageName, and the internals, get all pageDatas that use this 
+ * component as their entry point.
+ * @param virtualModulePrefix The prefix used to create the virtual module
+ * @param id Virtual module name
+ */
+export function getPagesFromVirtualModulePageName(internals: BuildInternals, virtualModulePrefix: string,  id: string): PageBuildData[]
+{
+	const path = getComponentFromVirtualModulePageName(virtualModulePrefix, id);
+
+	const pages: PageBuildData[] = [];
+	internals.pagesByKeys.forEach(pageData => {
+		if (pageData.component === path) {
+			pages.push(pageData);
+		}
+	});
```

Comment:
nit: can you use a `for..of` instead of a `forEach`?

Reply:
This is a Map object, on which `forEach` is typically recommended unlike with arrays